lib/events.js

```diff
@@ -73,46 +72,37 @@ class AsyncEvents extends EventEmitter {
    * // Emits an app event with a config arg
    * return app.events.emit('sector001', config);
    */
-  emit(...args) {
-    /*
+  emit(name, ...args) {
+    /**
      * Helper function that will always return a promise even if function is
      * synchronous and doesn't return a promise.
-     * @todo: this is very old kbox code, can we update it a bit?
+     *
+     * @param {Function} fn
+     * @param {...any} args
+     * @return {Promise}
      */
-    const handle = (...args) => {
-      const fn = args.shift();
+    const handle = (fn, ...args) => {
       const result = fn.apply(this, args);
       return Promise.resolve(result);
     };
     // Save for later
     const self = this;
-    // Grab name of event from first argument.
-    const name = args.shift();
 
-    // Grab priority sorted listeners for this event
-    const evnts = _(this._listeners)
-      // Filter by name
+    const fns = this._listeners
       .filter(listener => listener.name === name)
-      // Sort by priority
-      .sortBy('priority')
-      // Return value
-      .value();
-
-    // Map to array of func
-    const fns = _.map(evnts, evnt => evnt.fn);
+      .sort((a, b) => a.priority - b.priority)
+      .map(evnt => evnt.fn);
 
     // Log non engine events so we can keep things quiet
-    if (!_.includes(name, '-engine-')) {
+    if (!name.includes('-engine-')) {
       this.log.debug('emitting event %s', name);
       this.log.silly('event %s has %s listeners', name, fns.length);
     }
 
     // Make listener functions to a promise in series.
     return Promise.each(fns, fn => {
```

Comment:
If we ever get rid of BlueBird, this can be refactored to

```js
    for (const fn of fns) {
      const fnArgs = [...args];
      await fn(...fnArgs);
    }

    return !!fn.length;
```